lib/ansible/module_utils/ansible_tower.py

```diff
@@ -47,7 +49,6 @@ def tower_auth_config(module):
     '''
     config_file = module.params.pop('tower_config_file', None)
     if config_file:
-        config_file = os.path.expanduser(config_file)
```

Comment:
This doesn't look particularly wrong... if someone gives a filename with `~` in it, there shouldn't be a problem with expanding the user directory. Maybe not a necessary feature, but I don't think removing it will address any issue.

Below, however, I still see

```python
return parser.string_to_dict(f.read())
```

The file contents are in the Python config-file format (with the weird exception that it doesn't have the "[general]" header). That's a _big_ issue right now. If this is addressing any problem with the config file parsing I would like for it to address that, unless I haven't understood the main goal of this PR.

Reply:
> This doesn't look particularly wrong... if someone gives a filename with ~ in it, there shouldn't be a problem with expanding the user directory. Maybe not a necessary feature, but I don't think removing it will address any issue.

`os.path.expanduser` and `os.path.expandvars` are [already called](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1988) because [type of `tower_config_file` is `path`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ansible_tower.py#L91).

> Below, however, I still see
> 
>     return parser.string_to_dict(f.read())
>
> The file contents are in the Python config-file format (with the weird exception that it doesn't have the "[general]" header).

This is not what [documentation of Tower modules](https://docs.ansible.com/ansible/latest/modules/tower_project_module.html#notes) states:  "Python config-file format" isn't used, a custom format is.

> That's a big issue right now. If this is addressing any problem with the config file parsing I would like for it to address that, unless I haven't understood the main goal of this PR.

While creating this PR I missed the fact that tower-cli default paths were already taken in account. File format of `tower_config_file` isn't the same than the one used for tower-cli configuration file but this isn't an issue per se. Then I kept only the first 'cleaning' commits and I updated title and description of this PR.

I am not convinced that `tower_config_file` should use another file format since I would rather remove this parameter:
- `/etc/tower/tower_cli.cfg`, `~/.tower_cli.cfg`, `.tower_cli.cfg` or environment variables could be used instead of this parameter
- `tower_config_file` parameter seems a workaround for *`tower-cli` doesn't allow to use a custom config path* (which should be fixed in `tower-cli`).

Reply:
thanks for the detailed response.

Now I'm not happy with the documentation, because the filename strongly suggests parity with the tower-cli config file.

You're not wrong that tower-cli should accept a custom config path, it was just never on the radar, so here's an issue for that https://github.com/ansible/tower-cli/issues/577